services/galley/src/Galley/API/Query.hs

```diff
@@ -147,6 +154,47 @@ getUnqualifiedConversation lusr cnv = do
   c <- getConversationAndCheckMembership (tUnqualified lusr) (qualifyAs lusr cnv)
   Mapping.conversationView lusr c
 
+getGlobalTeamConversation ::
+  Members
+    '[ ConversationStore,
+       ErrorS 'NotATeamMember,
+       Error InternalError,
+       MemberStore,
+       TeamStore
+     ]
+    r =>
+  Local UserId ->
+  ClientId ->
+  TeamId ->
+  Sem r Public.GlobalTeamConversation
+getGlobalTeamConversation lusr cid tid = do
+  let uid = tUnqualified lusr
+  void $ noteS @'NotATeamMember =<< E.getTeamMember tid (tUnqualified lusr)
+  E.getGlobalTeamConversation tid >>= \case
+    Nothing -> do
+      conv <- E.createGlobalTeamConversation (qualifyAs lusr tid) uid
+      mlsData <- case Conv.convProtocol conv of
+        ProtocolMLS mls -> pure mls
+        ProtocolProteus -> throw (InternalErrorWithDescription "Wrong protocol, expected MLS, got Proteus.")
+      -- FUTUREWORK: remove this. we are planning to remove the need for a nullKeyPackageRef
+      let convId = Conv.convId conv
+          lconv = qualifyAs lusr convId
+      E.addMLSClients lconv (qUntagged lusr) (Set.singleton (cid, nullKeyPackageRef))
```

Comment:
Don't actually create the MLS clients here. Instead add logic in the commit handling code to deal with the global team conversation special case. (Wait for refactoring to be merged, #2844 )